File: frontend/src/app/services/user-dto.ts
export class UserDTO {
  id: number;
  name: string;
  surname: string;
  email: string;
  username: string;

  constructor(id: number, name: string, surname: string, email: string, username: string) {
    this.id = id;
    this.name = name;
    this.surname = surname;
    this.email = email;
    this.username = username;
  }
}

File: backend/src/main/java/pl/dmcs/springbootjsp_iwa/dto/UserDTO.java
package pl.dmcs.springbootjsp_iwa.dto;

public class UserDTO {
    private Long id;
    private String name;
    private String surname;
    private String email;
    private String username;

    public UserDTO(Long id, String name, String surname, String email, String username) {
        this.id = id;
        this.name = name;
        this.surname = surname;
        this.email = email;
        this.username = username;
    }

    // Getters and setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSurname() {
        return surname;
    }

    public void setSurname(String surname) {
        this.surname = surname;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }
}
